### PR TITLE
Refactor password recovery mutations

### LINKED ISSUE #4153: Refactoring of password recovery system
Proposed changes:

- we don't have mutations for password recovery and account activation
- mutation which reset password should also return JWT token so user don't need to pass login/password twice (usually after password change user want to access page)
- for password recovery we are encoding user pk (`urlsafe_base64_encode(force_bytes(user.pk))`) which is not good practice (exposing database ids ). Much better would be using account email for that.
- mutation for password reset should have url for recovery view as additional field - GraphQL API does not know how routing in PWA looks like so backend cant ATM attach valid URL ino emai

Contact me if you need more background for why we need to include those changes

### PR DESCRIPTION
**Pull request by fowczarek**
I want to merge this change because resolve #4153

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [x] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for fc5217ebe44464d58915ebf594463b2656d5a216 (mirumee/saleor @ 4153/Reactor_password_recovery_mutations)
Base comparison is e96b57dba480dbf071830e90816e77df1c2044bd.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  add billing address to checkout     	         34	         34	             20
  add shipping to checkout            	          7	          7	              0
  checkout payment charge             	         14	         14	              0
  complete checkout                   	          6	          6	              0
  create checkout                     	         48	         48	             24

# api.benchmark homepage
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve main menu                  	          5	          5	              0
  retrieve product list               	          4	          4	              0
  retrieve secondary menu             	          5	          5	              0
  retrieve shop                       	          2	          2	              0

# api.benchmark product
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  product details                     	         13	         13	              3

# api.benchmark variant
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve variant list               	         15	         15	              8
```

</p>
</details>

File: saleor/graphql/account/mutations/base.py [hunk 1]
```diff
@@ -111,6 +113,15 @@ class Meta:
     def perform_mutation(cls, _root, info, **data):
         email = data["email"]
         redirect_url = data["redirect_url"]
+
+        parsed_url = urlparse(redirect_url)
```
Comment (NyanKiyoshi) [inline saleor/graphql/account/mutations/base.py]: Note: it raises a `ValueError` if the URL is invalid

File: saleor/graphql/account/mutations/base.py [hunk 2]
```diff
@@ -93,19 +98,35 @@ class Arguments:
             required=True,
             description="Email of the user that will be used for password recovery.",
         )
+        redirect_url = graphene.String(
+            required=True,
+            description=(
+                "Url of storefront view which allow user reset password, "
```
Comment (maarcingebala) [inline saleor/graphql/account/mutations/base.py]: I would reword this a little bit. We could also provide an example here of expected format using the example.com domain.

"URL of a view where users should be redirected to reset the password. URL in RFC 1808 format e.g. <example>"

Ideally, we should have our own `URL` scalar that we could use across the entire API that would validate if provided URL is valid.

File: saleor/graphql/account/mutations/base.py [hunk 3]
```diff
@@ -14,26 +16,23 @@
...
-    context = {
-        "email": user.email,
-        "uid": urlsafe_base64_encode(force_bytes(user.pk)),
-        "token": default_token_generator.make_token(user),
-        "site_name": site.name,
-        "domain": site.domain,
-        "protocol": "https" if settings.ENABLE_SSL else "http",
-    }
-    emails.send_password_reset_email.delay(context, user.email, user.pk)
+def send_user_password_reset_email(redirect_url, user):
```
Comment (maarcingebala) [inline saleor/graphql/account/mutations/base.py]: This function could be moved to `saleor/account/emails`. If we added the ability to trigger a password reset for customers from the dashboard, it could be reused. I would also add short docstring e.g. "Trigger sending a password reset email for the given user."

File: saleor/graphql/account/mutations/base.py [hunk 4]
```diff
@@ -93,19 +98,35 @@ class Arguments:
...
 
     class Meta:
         description = "Sends an email with the account password modification link."
 
     @classmethod
     def perform_mutation(cls, _root, info, **data):
         email = data["email"]
+        redirect_url = data["redirect_url"]
+
+        parsed_url = urlparse(redirect_url)
```
Comment (maarcingebala) [inline saleor/graphql/account/mutations/base.py]: It would be nice to extract this validation to a function, as we will want to validate the URL against `ALLOWED_STOREFRONT_HOSTS` in a few places. Not sure where we could put the function but maybe somewhere in `saleor/core`?

File: saleor/account/emails.py [hunk 5]
```diff
@@ -8,25 +11,45 @@
...
         from_email=settings.DEFAULT_FROM_EMAIL,
-        recipient_list=[recipient],
+        recipient_list=[recipient_email],
         context=context,
     )
     account_events.customer_password_reset_link_sent_event(user_id=user_id)
 
 
+@app.task
+def send_password_reset_email(context, recipient, user_id):
```
Comment (NyanKiyoshi) [inline saleor/account/emails.py]: `recipient_email` for consistency

File: saleor/core/utils/url.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+from django.http.request import validate_host
+
+
+def validate_storefront_url(url):
+    """Validate the storefront URL.
+
+    Raise ValidationError if URL isn't in RFC 1808 format
+    or it isn't allowed by ALLOWED_STOREFRONT_HOSTS in settings.
+    """
+    parsed_url = urlparse(url)
```
Comment (NyanKiyoshi) [inline saleor/core/utils/url.py]: Can raise `ValueError`
